packages/table/src/interactions/selectable.tsx

```diff
@@ -9,10 +9,11 @@ import { Utils as BlueprintUtils } from "@blueprintjs/core";
 import * as PureRender from "pure-render-decorator";
```

Comment:
refactored this whole file. see unit tests (in `selectableTests.tsx`) for a description of the behavior.